packages/story-editor/src/utils/isTargetCoveringContainer.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+function isTargetCoveringContainer(target, container) {
+  const targetBox = target.getBoundingClientRect();
+  const containerBox = container.getBoundingClientRect();
+
+  const intersectionArea =
+    Math.max(
+      0,
+      Math.min(targetBox.right, containerBox.right) -
+        Math.max(targetBox.left, containerBox.left)
+    ) *
+    Math.max(
+      0,
+      Math.min(targetBox.bottom, containerBox.bottom) -
+        Math.max(targetBox.top, containerBox.top)
+    );
+
+  const containerArea = containerBox.width * containerBox.height;
+  const coverRatio = intersectionArea / containerArea;
+
+  // We can also use unionRatio, the difference is: when the element is bigger
+  // than the container then coverRatio = 1, while unionRatio will be < 1 and only
+  // be 1 if they both are equal in size/position.
+  // const targetArea = targetBox.width * targetBox.height;
+  // const unionArea = targetArea + containerArea - intersectionArea;
+  // const unionRatio = intersectionArea / unionArea;
+  // return unionArea > 0.99;
+
+  return coverRatio > 0.995;
+}
```

Comment:
You can uncomment this code and check the alternative solution, right now it will be set as BG even if the image is bigger than canvas, alternatively, it can be set as BG only if all edges ~match (0.99+ is equal to resizing the image with the current edge snapping).

Reply:
> right now it will be set as BG even if the image is bigger than canvas

That's the intention here -- whenever it's bigger than canvas, we should set it.

The reason for this is that (some) users are not aware that they can set background and instead of setting a background and scaling, they resize media to cover the page and then resize it even more to scale it 😛